File: src/main/java/org/lanternpowered/server/network/vanilla/message/type/connection/MessageOutDisconnect.java
package org.lanternpowered.server.network.vanilla.message.type.connection;

import static com.google.common.base.Preconditions.checkNotNull;

import org.lanternpowered.server.network.message.Message;
import org.spongepowered.api.text.Text;

public final class MessageOutDisconnect implements Message {

    private final Text reason;

    /**
     * Creates a new disconnect message with a specific reason.
     * 
     * @param json the reason
     */
    public MessageOutDisconnect(Text reason) {
        this.reason = checkNotNull(reason, "reason");
    }

    /**
     * Gets the reason.
     * 
     * @return the reason
     */
    public Text getReason() {
        return this.reason;
    }
}

File: src/main/kotlin/org/lanternpowered/server/network/vanilla/message/type/connection/MessageOutDisconnect.kt
/*
 * Lantern
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) SpongePowered <https://www.spongepowered.org>
 * Copyright (c) contributors
 *
 * This work is licensed under the terms of the MIT License (MIT). For
 * a copy, see 'LICENSE.txt' or <https://opensource.org/licenses/MIT>.
 */
package org.lanternpowered.server.network.vanilla.message.type.connection

import org.lanternpowered.server.network.message.Message
import org.spongepowered.api.text.Text

/**
 * A message to send a disconnect reason to the client.
 *
 * @param reason The reason of the disconnection
 */
data class MessageOutDisconnect(val reason: Text) : Message
